src/expand.ts

```diff
@@ -169,7 +169,14 @@ async function expandStringHelper(tmpl: string, opts: ExpansionOptions) {
     while ((mat = env_re3.exec(tmpl))) {
         const full = mat[0];
         const varname = mat[1];
-        const repl = fixPaths(env[varname]) || '';
+        const repl: string = fixPaths(env[varname]) || '';
+        // Avoid replacing an env variable by itself, e.g. PATH:env{PATH}.
+        const env_re4 = RegExp(`\\$env\\{(${varValueRegexp})\\}`, "g");
+        mat = env_re4.exec(repl);
+        const varnameRepl = mat ? mat[1] : undefined;
+        if (varnameRepl && varnameRepl === varname) {
+            break;
```

Comment:
I think we just want to log here.  We already have a failsafe that stops the recursion, just make sure we don't log the same warning 10 times.

Reply:
@bobbrow we try to expand this 3 times, in `expandConfigurePreset` and `expandBuildPreset`, so although we won't see the error 10 times, but we will see it 3 times. IS that fien?

Reply:
That's ok.